File: java/src/main/java/api/cornell/AcademicCareersResponse.java
package api.cornell;

import api.cornell.data.classes.AcademicCareer;
import com.google.gson.annotations.SerializedName;
import org.jetbrains.annotations.Nullable;

import java.util.List;

/**
 * {@code AcademicCareersResponse} is the response of a Academic Careers request.
 */
final class AcademicCareersResponse {
    
    /**
     * Main data of response.
     */
    @Nullable
    private final Data data;
    
    /**
     * Constructor for GSON.
     */
    private AcademicCareersResponse() {
        data = null;
    }
    
    /**
     * Obtain the academic careers information.
     *
     * @return the academic careers information.
     */
    @Nullable
    public List<AcademicCareer> getAcademicCareers() {
        if (data == null) {
            return null;
        }
        return data.academicCareers;
    }
    
    /**
     * {@code Data} is an uninteresting class that just holds an array of {@code Page}.
     */
    private static class Data {
        
        @Nullable
        @SerializedName(value = "acadCareers")
        private List<AcademicCareer> academicCareers;
    
        /**
         * Constructor for GSON.
         */
        private Data() {
            academicCareers = null;
        }
        
    }
    
}

File: kotlin/src/main/kotlin/api/cornell/response/AcademicCareersResponse.kt
package api.cornell.response

import api.cornell.data.AcademicCareer
import com.google.gson.annotations.SerializedName

/**
 * [AcademicCareersResponse] is the response of a Academic Careers request.
 */
internal class AcademicCareersResponse private constructor() {

    /**
     * Main data of response.
     */
    private val data: Data? = null
    /**
     * Obtain the academic careers information.
     */
    val academicCareers: List<AcademicCareer>?
        get() = data?.academicCareers

    /**
     * [Data] is an uninteresting class that just holds an array of [AcademicCareer].
     */
    private inner class Data private constructor() {
        @SerializedName(value = "acadCareers")
        val academicCareers: List<AcademicCareer>? = null
    }

}

